### PR TITLE
[FormLayout] Increase spacing from 200 to 300

### LINKED ISSUE #10618: [FormLayout] Fix spacing between form elements
## Issue summary
We need to increase the space between form elements from 8px (200) to 16px (400).

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/10618

| [Before](storybook.polaris.shopify.com) | [After](https://5d559397bae39100201eedc1-qxxcgtraos.chromatic.com/?path=/story/all-components-formlayout--all) |
|--------|--------|
|Screenshot 2023-10-27 at 2 43 38 PM|Screenshot 2023-10-27 at 2 44 54 PM|

🌀 [Spinstance](https://admin.web.tophat-form-layout-refactor.chloe-rice.us.spin.dev/store/shop1/draft_orders/new)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by heyjoethomas — APPROVED** [review]
Thanks Chloe!

File: polaris-react/src/components/FormLayout/FormLayout.scss [hunk 1]
```diff
@@ -1,15 +1,16 @@
 @import '../../styles/common';
 
 $item-min-size: 220px;
+$item-spacing: var(--p-space-400);
```
Comment (johanstromqvist) [inline polaris-react/src/components/FormLayout/FormLayout.scss]: The revised value is `300`. Please see comment in [the issue](https://github.com/Shopify/polaris/issues/10618).

**Review by johanstromqvist — DISMISSED** [review]

**Review by FaycalSabaou — APPROVED** [review]

**Comment by johanstromqvist** [comment]
The `400` to `300` fix between items looks good!

`300` between Form Layouts is a bit too tight and cause equidistance inside an between groups. It should be `400` to separate the groups slightly and to make it clear that the title belongs to inputs below. image

One question, could the 300 left margin in FormLayout.Group cause problems? I assume it's primarily there to give space between each input in the group, but could the margin on the first child conflict with other spacing methods? E.g. unintentionally stack margins, or cause a 300 padding to the card edge instead of a 400?

**Comment by johanstromqvist** [comment]
A few things stick out when inspecting [the composition with all variants](https://5d559397bae39100201eedc1-fqjipehhlu.chromatic.com/?path=/story/all-components-formlayout--all), but I think these are issues in the composition and not in the component itself.

- The "form group help text" might be an anti-pattern. I can't remember seeing it in the admin. Is this something we're committed to or could it be avoided? Thoughts:
  - I wonder if it's helpful to merchants to get further information about multiple inputs. I'd expect such context to be provided under the card title or group title.
  - It's hard to get the spacing right so that it looks like it belongs to the group. Currently it's closer to the title below (12px) than to the group it belongs to (16px). This can be fixed by tightening up the space, but it would still be hard to see exactly what it represents. image</details> 
  - It also doesn't play nicely with help texts on individual inputs. Odd layouts can happen when both are present. imageimage

- The "form group help text" is also misaligned, 20px from the card edge instead of 16px. I assume the help text isn't a part of the FormLayout and therefor not important to fix, but it might imply that the FormLayout is error prone. From a builder POV it makes sense to wanna give the help text box a 16px left padding, but it then gets another 4px from the composition. This is unexpected and a bit hard to track down.

**Comment by chloerice** [comment]
@johanstromqvist Thank you for the thorough tophat! I've updated the Chromatic link in the PR description for tophatting what's currently in the branch:
- RE the screenshot with the alignment of the help text on the second input: it looks to be correct, the help text appears it was set on the second input not the first. I'm not seeing that help text in production Storybook or in my branch though to validate that assumption 👀 
- The group's help text is fixed. It is part of FormLayout.Group and is composed in a Box in Group.tsx so the spacing needed to be fixed there not in the SCSS. I believe groups have support for help text for cases where a group of inputs are interdependent and need to be described as a whole vs setting individual descriptions on the inputs, similar to the need for support for errors at the group level (this is why InlineError was added to the system, as CollectionDetails condition input groups are a unit)
    Screenshot 2023-10-27 at 4 06 16 PM

- I've refactored the FormLayout to use the layout foundations instead of the antiquated negative margin flex hack, which isn't valid anymore given `gap` has full browser support (this is why the new stacks don't use negative margin like LegacyStack does)
- I've made the spacing between Groups 400

One thing I was wondering about was the lack of hierarchy (besides spacing) of the group title, it blends in as regular body text instead of standing out enough to label the group. I tested making the group title font-weight medium instead of regular to see if that would help, adding a title to the Card so it could be compared to existing form hierarchy:

|FormLayout.Group font-weight| [Current](https://5d559397bae39100201eedc1-qxxcgtraos.chromatic.com/?path=/story/all-components-formlayout--all) (Regular) | [Proposed](https://5d559397bae39100201eedc1-vtrhtwsvzs.chromatic.com/?path=/story/all-components-formlayout--all) (Medium)|
|---|--------|--------|
|| Screenshot 2023-10-27 at 2 44 54 PM|Screenshot 2023-10-27 at 2 43 52 PM|

**Comment by johanstromqvist** [comment]
Thanks for the fixes and refactoring, @chloerice! 😍

Let's ship the spacing fixes!

Regarding the hierarchy/typography. I agree that it's off, but I'm pretty sure this is on Tjosé's mind (see what I did there). Lots of Gen 3 type improvements aren't codified and shipped yet AFAIK. However, shipping a quick fix would likely add value and seems low risk to me.

**Review by johanstromqvist — APPROVED** [review]

**Pull request merged**